src/framework/preferences/js/SyllabificationEnactor.js

```diff
@@ -222,6 +222,13 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
         }
 
         var src = fluid.stringTemplate(pattern, that.options.terms);
+        var existingLang = fluid.get(that, ["hyphenatorSRCs", src]);
```

Comment:
Rather than code around this problem by creating an extra piece of "back channel state" in the only function we have which has a hard dependency on Hypher, I think we should try to factor "fluid.prefs.enactor.syllabification.createHyphenator" a bit better.
I think we really have two problems in one - as we "smelled " there's indeed a race condition in luid.prefs.enactor.syllabification.waitForHyphenators that would very likely make the hard failure go away if we fixed it - that is, if it continued to iterate over freshly arrived promises . However, this would still end up with an inefficient component which made multiple network requests to the same file.

So what I suggest is, rather than keeping records of hyphenators at multiple paths in "that.hyphenators" that we simply mirror the real state at global Hypher.languages.xxx in our own. This means factoring out the fallback logic within getHyphenator into a self-contained function, such that we only get around to calling createHyphenator once we have already resolved the fallback.
This would result in a much better getHyphenator that 
i) Had only one site for calling createHyphenator
ii) Had only one site for assigning to that.hyphenators, which made only one assignment per call of getHyphenator/createHyphenator
iii) Maintained hyphenators keyed by "resolved language" rather than "requested language" meaning that we only had one entry in our cache for each pattern.